netbox/dcim/filters.py

```diff
@@ -420,6 +424,15 @@ def filter_type(self, queryset, name, value):
             return queryset.filter(form_factor=IFACE_FF_LAG)
         return queryset
 
+    def _mac_address(self, queryset, name, value):
+        value = value.strip()
+        if not value:
+            return queryset
+        try:
+            return queryset.filter(mac_address=value).distinct()
```

Comment:
`.distinct()` is unnecessary here. It's used in `DeviceFilter` to avoid returning duplicate Devices in the event a device has more than one interface with the same MAC address.